pkg/reconciler/ingress/resources/httproute.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/utils/pointer"
+	gwv1alpha1 "sigs.k8s.io/gateway-api/apis/v1alpha1"
+
+	"knative.dev/networking/pkg"
+	netv1alpha1 "knative.dev/networking/pkg/apis/networking/v1alpha1"
+	"knative.dev/pkg/kmeta"
+
+	"knative.dev/net-gateway-api/pkg/reconciler/ingress/config"
+)
+
+// MakeHTTPRoute creates HTTPRoute to set up routing rules.
+func MakeHTTPRoute(
+	ctx context.Context,
+	ing *netv1alpha1.Ingress,
+	rule *netv1alpha1.IngressRule,
+) (*gwv1alpha1.HTTPRoute, error) {
+
+	visibility := ""
+	if rule.Visibility == netv1alpha1.IngressVisibilityClusterLocal {
+		visibility = "cluster-local"
+	}
```

Comment:
I guess this is where the helper above would apply, I like this more personally.